packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -0,0 +1,765 @@
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import 'ink_well.dart';
+import 'material.dart';
+
+// TODO(clocksmith): Refactor since this is copied from ink_ripple.dart
+RectCallback? _getClipCallback(
+  RenderBox referenceBox,
+  bool containedInkWell,
+  RectCallback? rectCallback,
+) {
+  if (rectCallback != null) {
+    assert(containedInkWell);
+    return rectCallback;
+  }
+  if (containedInkWell) return () => Offset.zero & referenceBox.size;
+  return null;
+}
+
+// TODO(clocksmith): Refactor since this is copied from ink_ripple.dart
+double _getTargetRadius(
+  RenderBox referenceBox,
+  bool containedInkWell,
+  RectCallback? rectCallback,
+  Offset position,
+) {
+  final Size size =
+      rectCallback != null ? rectCallback().size : referenceBox.size;
+  final double d1 = size.bottomRight(Offset.zero).distance;
+  final double d2 =
+      (size.topRight(Offset.zero) - size.bottomLeft(Offset.zero)).distance;
+  return math.max(d1, d2) / 2.0;
+}
+
+  /// Begin an m3 sparkle ripple, centered at [position] relative to
```

Comment:
can you add documentation for all parameters and a short code sample showing usage?

Reply:
Also, should it be "a Material 3"?

Reply:
Done

Reply:
Since a proper sample relies on dartpad, I added instructions, and will add a proper example once https://github.com/flutter/flutter/issues/85238 is resolved